### PR TITLE
Fix driver worker

### PR DESCRIPTION
**Pull request by wssgcg1213**
## performance update
1. no more duplicated style element
   `<view a:for="moduleList"><module data="item" /></view>`
  each `style` tag in custom component: `module` will be dispatch from worker to renderer, and then append to DOM tree, which will case repaint and stuck scrolling even saw a flash of screen.
2. reduce size of mutationRecord, by removing undefined keys and remove useless childNodes (an obveriously bug)
3. remove all comment nodes, which is a big spend in transfor, add nodes and remove nodes.
4. others, like remove `"ns": null,` in attribute, reduce 10 chars every attr.

### CODE REVIEW

File: packages/driver-worker/src/WorkerDriver.js [hunk 1]
```diff
@@ -16,13 +17,18 @@ const TO_SANITIZE = [
...
     super(workerGlobalScope.document);
 
     this.evaluator = new Evaluator(postMessage);
     this.nodesMap = new Map();
     this.nodeCounter = 0;
+    /**
+     * Can enable some performance effect by judging redndererType.
+     *
+     */
+    this.redndererType = rendererType;
```
Comment (yuanyan) [inline packages/driver-worker/src/WorkerDriver.js]: 1. redndererType typo erorr 2. rendererType 不应该出现 driver 里，概念上有问题，从代码上也看不懂解决什么问题

File: packages/driver-worker/src/Driver.js [hunk 2]
```diff
@@ -53,7 +54,7 @@ export default class Driver {
   }
 
   createEmpty() {
-    return this.createComment(' empty ');
+    return this.createComment(EMPTY_CONTENT);
```
Comment (wssgcg1213) [inline packages/driver-worker/src/Driver.js]: 减小通信体积

File: packages/driver-worker/src/WorkerDriver.js [hunk 3]
```diff
@@ -37,7 +56,8 @@ export default class WorkerDriver extends Driver {
         let mutation = mutations[i];
         for (let j = TO_SANITIZE.length; j--;) {
           let prop = TO_SANITIZE[j];
-          mutation[prop] = this.sanitize(mutation[prop], prop);
+          const value = this.sanitize(mutation[prop], prop);
+          if (value) mutation[prop] = value;
```
Comment (wssgcg1213) [inline packages/driver-worker/src/WorkerDriver.js]: 减小消息体积 (undefined 值不发送)

File: packages/driver-worker/src/WorkerDriver.js [hunk 4]
```diff
@@ -93,13 +120,17 @@ export default class WorkerDriver extends Driver {
...
+          if (this.deduplicateStyle && node.nodeName === STYLE_ELEMENT) {
+            const textStyle = getChildText(node);
+            if (this.hitStyle[textStyle]) return null;
+            this.hitStyle[textStyle] = node;
+          }
+
+          const events = node._getEvents();
+          if (events.length > 0) result.events = events;
+          if (node.attributes) result.attributes = node.attributes;
+          if (Object.keys(node.style).length > 0) result.style = node.style;
```
Comment (wssgcg1213) [inline packages/driver-worker/src/WorkerDriver.js]: 减小消息体积 (空值不发送)

File: packages/driver-worker/src/WorkerDriver.js [hunk 5]
```diff
@@ -93,13 +120,17 @@ export default class WorkerDriver extends Driver {
...
-            nodeName: node.nodeName,
-            style: node.style,
-            childNodes: node.childNodes && node.childNodes.map((node) => this.sanitize(node, prop)),
-          });
+          result.nodeName = node.nodeName;
+          if (this.deduplicateStyle && node.nodeName === STYLE_ELEMENT) {
+            const textStyle = getChildText(node);
+            if (this.hitStyle[textStyle]) return null;
+            this.hitStyle[textStyle] = node;
+          }
```
Comment (wssgcg1213) [inline packages/driver-worker/src/WorkerDriver.js]: 不重复下发字符串匹配的 style 节点

**Comment by wssgcg1213** [comment]
driver-worker All files coverage: 58.25%

**Pull request merged**